pkg/hub/managedclusterset/controller.go

```diff
@@ -165,35 +98,50 @@ func (c *managedClusterSetController) syncClusterSet(ctx context.Context, cluste
 		emptyCondition.Reason = "ClustersSelected"
 		emptyCondition.Message = fmt.Sprintf("%d ManagedClusters selected", count)
 	}
-	meta.SetStatusCondition(&clusterSetCopy.Status.Conditions, emptyCondition)
+	meta.SetStatusCondition(&clusterSet.Status.Conditions, emptyCondition)
 
 	// skip update if cluster set status does not change
-	if reflect.DeepEqual(clusterSetCopy.Status.Conditions, clusterSet.Status.Conditions) {
+	if reflect.DeepEqual(clusterSet.Status.Conditions, originalClusterSet.Status.Conditions) {
 		return nil
 	}
 
-	_, err = c.clusterClient.ClusterV1beta1().ManagedClusterSets().UpdateStatus(ctx, clusterSetCopy, metav1.UpdateOptions{})
+	_, err = c.clusterClient.ClusterV1beta1().ManagedClusterSets().UpdateStatus(ctx, clusterSet, metav1.UpdateOptions{})
 	if err != nil {
-		return fmt.Errorf("failed to update status of ManagedClusterSet %q: %w", clusterSetCopy.Name, err)
+		return fmt.Errorf("failed to update status of ManagedClusterSet %q: %w", clusterSet.Name, err)
 	}
 
 	return nil
 }
 
-func getClusterSet(obj interface{}) string {
-	accessor, err := meta.Accessor(obj)
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("error to get accessor of object: %v", obj))
-		return ""
-	}
-	labels := accessor.GetLabels()
-	if len(labels) == 0 {
-		return ""
-	}
+func matchedClustersNumber(clusterLister clusterlisterv1.ManagedClusterLister, clusterSet *clusterv1beta1.ManagedClusterSet) (int, error) {
+	var selector labels.Selector
+	var err error
+	var clusters []*v1.ManagedCluster
+	selectorType := clusterSet.Spec.ClusterSelector.SelectorType
 
-	if clusterSetName := labels[clusterSetLabel]; len(clusterSetName) > 0 {
-		return clusterSetName
+	switch {
+	case selectorType == "" || selectorType == clusterv1beta1.LegacyClusterSetLabel:
+		selector = labels.SelectorFromSet(labels.Set{
+			clusterSetLabel: clusterSet.Name,
+		})
+
+		clusters, err = clusterLister.List(selector)
```

Comment:
if the diff is only selector, we should only generate selector in switch

Reply:
done